i-VRESSE/haddock3-webapp
# Scenario ligand + scenario dna/rna

Refs #29 

![0 0 0 0_3000_scenarios_protein-ligand (1)](https://github.com/i-VRESSE/haddock3-webapp/assets/1713488/2e05fe58-0ec6-4db4-8674-674465cd7ef9)

This PR also 
- upgrades [valibot](https://valibot.dev/)  as docs only show latest version and latest version is a [major rewrite](https://valibot.dev/blog/valibot-v0.31.0-is-finally-available/).
- makes sure each scenario page can only be submitted when residues are selected in both molecules.

To test:

1. Checkout this branch
2. Spin up deployment 
3. login
4. Goto http://0.0.0.0:3000/scenarios/protein-ligand 
5. as protein upload https://github.com/haddocking/haddock3/blob/main/examples/docking-protein-ligand/data/neuraminidase-2BAT.pdb 
6. as ligand upload https://github.com/haddocking/haddock3/blob/main/examples/docking-protein-ligand/data/oseltamivir_zwitterion.pdb
7. Select residues on both sides
8. Submit
9. The form should be valid and the job should complete successfully.

Things to test
1. PDB with multiple ligands, see how ligand selection is handled
2. Without custom ligand param and top files
3. Pages with valibot validation
4. Zero selected residues
5. Check that produced files makes sense, especially ambig files and resdic parameters
